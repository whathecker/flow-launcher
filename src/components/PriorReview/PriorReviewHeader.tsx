import React from "react";
import { StyleSheet, Image } from "react-native";
import { View, Text } from "../Themed";
import { GoBackButton } from "../shared";
import { Container, Typography } from "../../styles";
import { navigationRef } from "../../utils";

type PriorReviewHeaderProps = {
  title: string;
  motivation: string;
};

const PriorReviewHeader: React.FC<PriorReviewHeaderProps> = ({
  title,
  motivation,
}: PriorReviewHeaderProps) => {
  return (
    <View style={styles.wrapper}>
      <View style={styles.goBackBtnAreaWrapper}>
        <View style={styles.goBackBtnWrapper}>
          <GoBackButton pressHandler={() => navigationRef.goBack()} />
        </View>
      </View>
      <View style={styles.headerAreaWrapper}>
        <View style={styles.headerWrapper}>
          <Text style={styles.headerText}>{`REVIEW PRIORITY`}</Text>
          <Image
            style={styles.image}
            source={require(`../../../assets/images/bar-chart_1f4ca.png`)}
          />
        </View>
      </View>
      <View style={styles.goalAreaWrapper}>
        <Text style={styles.goalText}>{title}</Text>
        <Text style={styles.motivationText}>{motivation}</Text>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  wrapper: {
    paddingBottom: 15,
  },
  goBackBtnAreaWrapper: {
    ...Container.flexStart,
    height: "40%",
  },
  goBackBtnWrapper: {
    paddingLeft: 25,
    paddingTop: 0,
  },
  headerAreaWrapper: {
    paddingLeft: "10%",
  },
  headerWrapper: {
    ...Container.flexStart,
    width: "100%",
  },
  headerText: {
    ...Typography.h4,
    fontSize: 25,
    paddingRight: 10,
  },
  image: {
    width: 32,
    height: 32,
  },
  goalAreaWrapper: {
    paddingLeft: "10%",
    marginTop: "8%",
  },
  goalText: {
    ...Typography.h4,
    fontSize: 24,
    paddingBottom: 5,
  },
  motivationText: {
    ...Typography.p,
    fontSize: 18,
  },
});

export default PriorReviewHeader;
