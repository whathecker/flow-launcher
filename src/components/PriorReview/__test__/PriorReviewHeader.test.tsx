import * as React from "react";
import renderer from "react-test-renderer";
import PriorReviewHeader from "../PriorReviewHeader";

describe("Test PriorReviewHeader component", () => {
  it("Should renders PriorReviewHeader correctly", () => {
    const tree = renderer
      .create(
        <PriorReviewHeader title="Mock goal" motivation="Mock motivation" />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
