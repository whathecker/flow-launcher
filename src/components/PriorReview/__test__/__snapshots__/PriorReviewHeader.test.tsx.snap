// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PriorReviewHeader component Should renders PriorReviewHeader correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "paddingBottom": 15,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": "40%",
          "justifyContent": "flex-start",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
          },
          Object {
            "paddingLeft": 25,
            "paddingTop": 0,
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 20,
              "width": 20,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "paddingLeft": "10%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
          },
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "width": "100%",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
            },
            Object {
              "fontFamily": "EB-Garamond",
              "fontSize": 25,
              "paddingRight": 10,
            },
          ]
        }
      >
        REVIEW PRIORITY
      </Text>
      <Image
        source={1}
        style={
          Object {
            "height": 32,
            "width": 32,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "marginTop": "8%",
          "paddingLeft": "10%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
          },
          Object {
            "fontFamily": "EB-Garamond",
            "fontSize": 24,
            "paddingBottom": 5,
          },
        ]
      }
    >
      Mock goal
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
          },
          Object {
            "fontFamily": "Forum-Regular",
            "fontSize": 18,
          },
        ]
      }
    >
      Mock motivation
    </Text>
  </View>
</View>
`;
